src/workerd/api/http.h

```diff
@@ -632,6 +632,48 @@ struct RequestInitializerDict {
   });
 };
 
+class CfProperty {
```

Comment:
Non-blocking nit: since we're pulling this out, I have a mild preference towards separating new bits like this out into a separate h/c++ file if possible.